src/features/class-open-graph.php

```diff
@@ -0,0 +1,193 @@
+	 *
+	 * @return void
+	 */
+	public function add_post_type_support() {
+		add_post_type_support( 'post', 'open-graph' );
+		add_post_type_support( 'page', 'open-graph' );
+	}
+
+
+	/**
+	 * Enqueue block editor assets.
+	 *
+	 * @return void
+	 */
+	public function enqueue_block_editor_assets() {
+		global $post;
+
+		if ( ! is_admin()
+			|| ! ( $post instanceof \WP_Post || is_int( $post ) || is_null( $post ) )
+		) {
+			return;
+		}
+
+		$post_type = get_post_type( $post );
+
+		if ( empty( $post_type ) || ! post_type_supports( $post_type, 'open-graph' ) ) {
+			return;
+		}
+
+		wp_enqueue_script( 'wp-seo-open-graph-js' );
+	}
+
+	/**
+	 * Add meta fields.
+	 */
+	public function add_meta_fields(): void {
+		register_meta_helper(
+			'post',
+			get_post_types_by_support( 'open-graph' ),
+			'wp_seo_open_graph_title',
+			[
+				'sanitize_callback' => 'wp_kses_post',
+				'single'            => true,
+				'type'              => 'string',
+				'show_in_rest'      => true,
+			]
+		);
+
+		register_meta_helper(
+			'post',
+			get_post_types_by_support( 'open-graph' ),
+			'wp_seo_open_graph_description',
+			[
+				'sanitize_callback' => 'wp_kses_post',
+				'single'            => true,
+				'type'              => 'string',
+				'show_in_rest'      => true,
+			]
+		);
+
+		register_meta_helper(
+			'post',
+			get_post_types_by_support( 'open-graph' ),
+			'wp_seo_open_graph_image',
+			[
+				'sanitize_callback' => 'wp_kses_post',
+				'single'            => true,
+				'type'              => 'integer',
+				'show_in_rest'      => true,
+			]
+		);
+	}
+
+	/**
+	 * Get the title with a fallback.
+	 *
+	 * @param int $post_id The post ID.
+	 *
+	 * @return string The title.
+	 */
+	public static function get_title( $post_id ): string {
+		$open_graph_title = get_post_meta( $post_id, 'wp_seo_open_graph_title', true );
+
+		if ( ! empty( $open_graph_title ) && is_string( $open_graph_title ) ) {
+			return $open_graph_title;
+		}
+
+		return get_the_title( $post_id );
+	}
+
+	/**
+	 * Get the description with a fallback.
+	 *
+	 * @param int $post_id The post ID.
+	 *
+	 * @return string The description.
+	 */
+	public static function get_description( $post_id ): string {
+		$open_graph_description = get_post_meta( $post_id, 'wp_seo_open_graph_description', true );
+
+		if ( ! empty( $open_graph_description ) && is_string( $open_graph_description ) ) {
+			return $open_graph_description;
+		}
+
+		return get_the_excerpt( $post_id );
+	}
+
+	/**
+	 * Get the image with a fallback.
+	 *
+	 * @param int $post_id The post ID.
+	 *
+	 * @return string|false The image URL or false if no assigned images.
+	 */
+	public static function get_image( $post_id ): string|bool {
+		$open_graph_image = get_post_meta( $post_id, 'wp_seo_open_graph_image', true );
+
+		if ( ! empty( $open_graph_image ) && is_numeric( $open_graph_image ) ) {
+			return wp_get_attachment_image_url( (int) $open_graph_image, 'full' );
```

Comment:
would probably be safest to get the url, then check if it's set and not a wp_error, then return it. The media could get deleted from the media library after being set in the meta.

Reply:
Good call. I've made that update in [4d80003](https://github.com/alleyinteractive/wp-seo/pull/132/commits/4d8000396f15fecaed38aa9b241c654843a3f935)

Reply:
Additional update in 4d9b48426377677fec0f19a508eacc4d9d4caf91